app/webpacker/controllers/flatpickr_controller.js

```diff
@@ -0,0 +1,110 @@
+// import Flatpickr
+import Flatpickr from "stimulus-flatpickr";
+import { ar } from "flatpickr/dist/l10n/ar";
+import { cat } from "flatpickr/dist/l10n/cat";
+import { cy } from "flatpickr/dist/l10n/cy";
+import { de } from "flatpickr/dist/l10n/de";
+import { fr } from "flatpickr/dist/l10n/fr";
+import { it } from "flatpickr/dist/l10n/it";
+import { nl } from "flatpickr/dist/l10n/nl";
+import { pl } from "flatpickr/dist/l10n/pl";
+import { pt } from "flatpickr/dist/l10n/pt";
+import { ru } from "flatpickr/dist/l10n/ru";
+import { sv } from "flatpickr/dist/l10n/sv";
+import { tr } from "flatpickr/dist/l10n/tr";
+import { en } from "flatpickr/dist/l10n/default.js";
+import ShortcutButtonsPlugin from "shortcut-buttons-flatpickr";
+import labelPlugin from "flatpickr/dist/plugins/labelPlugin/labelPlugin";
+
+export default class extends Flatpickr {
+  static values = { enableTime: Boolean, mode: String };
+  static targets = ["start", "end"];
+  locales = {
+    ar: ar,
+    cat: cat,
+    cy: cy,
+    de: de,
+    fr: fr,
+    it: it,
+    nl: nl,
+    pl: pl,
+    pt: pt,
+    ru: ru,
+    sv: sv,
+    tr: tr,
+    en: en,
+  };
+
+  initialize() {
+    const datetimepicker = this.enableTimeValue == true;
+    const mode = this.modeValue == "range" ? "range" : "single";
+    // sets your language (you can also set some global setting for all time pickers)
+    this.config = {
+      altInput: true,
+      altFormat: datetimepicker
+        ? Spree.translations.flatpickr_datetime_format
+        : Spree.translations.flatpickr_date_format,
+      dateFormat: datetimepicker ? "Y-m-d H:i" : "Y-m-d",
+      enableTime: datetimepicker,
+      time_24hr: datetimepicker,
+      locale: I18n.base_locale,
+      plugins: this.plugins(mode, datetimepicker),
+      mode,
+    };
+  }
+
+  clear(e) {
+    this.fp.setDate(null);
+  }
+
+  change(selectedDates, dateStr, instance) {
+    if (this.hasStartTarget && this.hasEndTarget) {
+      this.startTarget.value = selectedDates[0]
+        ? this.fp.formatDate(selectedDates[0], this.config.dateFormat)
+        : "";
+      this.endTarget.value = selectedDates[1]
+        ? this.fp.formatDate(selectedDates[1], this.config.dateFormat)
+        : "";
+      // Also, send event to be sure that ng-model is well updated
+      this.startTarget.dispatchEvent(new Event("change"));
+      this.endTarget.dispatchEvent(new Event("change"));
+    }
+  }
+
+  // private
+
+  plugins = (mode, datetimepicker) => {
+    const buttons = [{ label: Spree.translations.close }];
+    if (mode == "single") {
+      buttons.unshift({
+        label: datetimepicker
+          ? Spree.translations.now
+          : Spree.translations.today,
+      });
+    }
+    return [
+      ShortcutButtonsPlugin({
+        button: buttons,
+        label: mode != "range" && "or",
```

Comment:
My JS knowledge is so outdated, I don't even know what this expression does.

Reply:
```
if mode != "range"
  label = "or"
else
  label = undefined
end
```

Actually this evaluate the expression and return the value of the last element of the logical expression

Reply:
Okay, that's actually what I thought but it didn't make much sense to me. What is the label `or` doing and why `undefined` otherwise?

Reply:
Now I read the documentation of the buttons plugin and the code a bit more and I understand the *why*. I'm not sure we need the "or" and if we need it then we should translate it.

But I also noticed a lot of if-else branches around what kind of data we are selecting and I'm wondering if it would be better to store each configuration in a separate object and then apply that generically.

Reply:
This `or` wording seems to be unnecessary, you're absolutely right:

##### with
<img width="357" alt="Capture d’écran 2022-09-05 à 15 17 55" src="https://user-images.githubusercontent.com/296452/188458816-e6fd5aef-fd8e-4b30-81d8-f868c81455e4.png">

##### without
<img width="354" alt="Capture d’écran 2022-09-05 à 15 17 20" src="https://user-images.githubusercontent.com/296452/188458703-24ea85f2-a72a-4ce9-b504-5919eb4bb648.png">

I'd vote to remove it, see commit 3a72c4e716